### DISCUSSION TITLE
How use vue3-json-viewer as plugin?

### DISCUSSION THREAD

**Discussion by SergioEanX** [nuxt/nuxt#18470] [Questions]
I installed vue3-json-viewer as plugin by creating a file `json-viewer.client.js` inside plugins folder with the following code:
```
import JsonViewer from 'vue3-json-viewer';
// if you used v1.0.5 or latster ,you should add import "vue3-json-viewer/dist/index.css"
import 'vue3-json-viewer/dist/index.css';
export default defineNuxtPlugin((nuxtApp) => {
  nuxtApp.vueApp.use(JsonViewer);
});
```
Then in my component I use it as:

`  <JsonViewer :value="metadata_json"></JsonViewer>`

It actually work but in my IDE  (VS Code) terminal I get the following warning:

[Vue warn]: Failed to resolve component: JsonViewer                                                                               09:02:34
If this is a native custom element, make sure to exclude it from component resolution via compilerOptions.isCustomElement.
[Vue warn]: Component <Anonymous> is missing template or render function.  

How I can fix it?

**Comment by TheAlexLichter** [comment]
If you initialize the plugin as `.client.js`, you can also use the component only on the client side. So wrap it in `<ClientOnly>` or create a `.client.vue` wrapper component

**Reply by SergioEanX** [reply]
Thanks! I was able to fix the warning with your suggestion:
```
 <ClientOnly>
      <JsonViewer :value="metadata_json">
      </JsonViewer>
 </ClientOnly>
```
